src/Microsoft.ML.EntryPoints/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,411 @@
+                "Mismatch between number of feature slots and number of features permuted.");
+
+            IEnumerable<BinaryMetrics> metrics = Enumerable.Empty<BinaryMetrics>();
+            for (int i = 0; i < permutationMetrics.Length; i++)
+            {
+                var pMetric = permutationMetrics[i];
+                metrics = metrics.Append(new BinaryMetrics
+                {
+                    FeatureName = slotNames[i],
+                    AreaUnderRocCurve = pMetric.AreaUnderRocCurve.Mean,
+                    AreaUnderRocCurveStdErr = pMetric.AreaUnderRocCurve.StandardError,
+                    Accuracy = pMetric.Accuracy.Mean,
+                    AccuracyStdErr = pMetric.Accuracy.StandardError,
+                    PositivePrecision = pMetric.PositivePrecision.Mean,
+                    PositivePrecisionStdErr = pMetric.PositivePrecision.StandardError,
+                    PositiveRecall = pMetric.PositiveRecall.Mean,
+                    PositiveRecallStdErr = pMetric.PositiveRecall.StandardError,
+                    NegativePrecision = pMetric.NegativePrecision.Mean,
+                    NegativePrecisionStdErr = pMetric.NegativePrecision.StandardError,
+                    NegativeRecall = pMetric.NegativeRecall.Mean,
+                    NegativeRecallStdErr = pMetric.NegativeRecall.StandardError,
+                    F1Score = pMetric.F1Score.Mean,
+                    F1ScoreStdErr = pMetric.F1Score.StandardError,
+                    AreaUnderPrecisionRecallCurve = pMetric.AreaUnderPrecisionRecallCurve.Mean,
+                    AreaUnderPrecisionRecallCurveStdErr = pMetric.AreaUnderPrecisionRecallCurve.StandardError
+                });
+            }
+
+            var result = mlContext.Data.LoadFromEnumerable(metrics);
+            return result;
+        }
+
+        private static IDataView GetMulticlassMetrics(
+            MLContext mlContext,
+            ISingleFeaturePredictionTransformer<object> predictor,
+            IDataView data,
+            PermutationFeatureImportanceArguments input)
+        {
+            var slotNames = GetSlotNames(data);
+
+            var permutationMetrics = mlContext.MulticlassClassification
+                .PermutationFeatureImportance(predictor,
+                                              data,
+                                              labelColumnName: input.LabelColumnName,
+                                              useFeatureWeightFilter: input.UseFeatureWeightFilter,
+                                              numberOfExamplesToUse: input.NumberOfExamplesToUse,
+                                              permutationCount: input.PermutationCount);
+
+            Contracts.Assert(slotNames.Length == permutationMetrics.Length,
+                "Mismatch between number of feature slots and number of features permuted.");
+
+            IEnumerable<MulticlassMetrics> metrics = Enumerable.Empty<MulticlassMetrics>();
```

Comment:
You should use a `List<MulticlassMetrics>` instead. Appending to Enumerable.Empty is quite inefficient.